### DISCUSSION TITLE
Add custom buttons to the right-click menu

### DISCUSSION THREAD

**Discussion by minihopp** [desjarlais/Scintilla.NET#67] [Q&A] [answered]
Hi,

I have implemented this and its very nice. 
Is there a way to add any custom buttons to the right-click menu?

**Comment by VPKSoft** [comment, marked as answer]
Hi,
Only way I know is to replace the default menu with a new context menu and re-implement the default functionality to a new custom ContextMenuStrip and add the additional menu items there.
![image](https://user-images.githubusercontent.com/40712699/234597164-3aa48e9a-2fe7-414f-bb33-ba9e48a8d0f2.png)

Just need to assign the new ContextMenuStrip to the Scintilla:
```cs
scintilla.ContextMenuStrip = contextMenuStrip1;
```
*The menu strip in the designer:*
![image](https://user-images.githubusercontent.com/40712699/234597678-b55a7f8a-bf8d-43e9-b818-d75751eb6e4f.png)

The code in the [ScriptNotepad](https://github.com/VPKSoft/ScriptNotepad) has an implementation (just scratch the excess stuff): https://github.com/VPKSoft/ScriptNotepad/blob/master/ScriptNotepad/UtilityClasses/ScintillaNETUtils/ScintillaContextMenu.cs

**Comment by minihopp** [comment]
I did ended up created my own ContextMenuStrip.
